components/log-viewer-webui/server/src/plugins/MongoReplicaServerPlugin.ts

```diff
@@ -0,0 +1,272 @@
+/* eslint-disable max-classes-per-file */
+import {FastifyInstance} from "fastify";
+import {
+    MongoClient,
+    MongoServerError
+} from "mongodb";
+import {Server} from "socket.io";
+
+
+/**
+ * Initialize the MongoDB replica set.
+ *
+ * @param {FastifyInstance} fastify
+ * @return {Promise<void>}
+ * @throws {Error} If the replica set initialization fails.
+ */
+const initializeReplicaSet = async (fastify: FastifyInstance): Promise<void> => {
+    try {
+        const directMongoClient = new MongoClient(
+            "mongodb://localhost:27017",
+            {replicaSet: "rs0", directConnection: true}
+        );
+        const response = await directMongoClient.db("admin").admin()
+            .command({replSetInitiate: {}});
+
+        fastify.log.info("Replica set initialized:", response);
+    } catch (e) {
+        if (e instanceof MongoServerError && "AlreadyInitialized" === e.codeName) {
+            return;
+        }
+        throw new Error("Failed to initialize replica set", {cause: e});
+    }
+};
```

Comment:
We can remove all references to this, since the clp-package should have done the init in some Python script when starting the package.
```suggestion
/**
 * Initialize the MongoDB replica set.
 *
 * @param {FastifyInstance} fastify
 * @return {Promise<void>}
 * @throws {Error} If the replica set initialization fails.
 */
const initializeReplicaSet = async (fastify: FastifyInstance): Promise<void> => {
    try {
        const directMongoClient = new MongoClient(
            "mongodb://localhost:27017",
            {replicaSet: "rs0", directConnection: true}
        );
        const response = await directMongoClient.db("admin").admin()
            .command({replSetInitiate: {}});

        fastify.log.info("Replica set initialized:", response);
    } catch (e) {
        if (e instanceof MongoServerError && "AlreadyInitialized" === e.codeName) {
            return;
        }
        throw new Error("Failed to initialize replica set", {cause: e});
    }
};
```